### PR TITLE
Add support for RS256, RS384, RS512 constants

### PR DESCRIPTION
**Pull request by hslatman**
This PR replaces https://github.com/veraison/go-cose/pull/140. It closes https://github.com/veraison/go-cose/issues/141.

It only provides the constants for the algorithms. A `cose.Signer` or `cose.Verifier` needs to be implemented and provided from outside the package to use these algorithms. Using `NewSigner` and `NewVerifier` will return an error when called with one of these algorithms, because PKCS#1 v1.5 signing and verification aren't implemented in `go-cose` itself.

Example usage with external `mycose` package:
```golang
import (
        "crypto/rand"

	"github.com/veraison/go-cose"
	"go.step.sm/crypto/keyutil"
	
	"mycose"
)

func main(){
	key, _ := keyutil.GenerateSigner("RSA", "", 2048)
	cs := mycose.NewRS256Signer(key)  // adheres to cose.Signer

        data := []byte("1234")
	sig, _ := cs.Sign(rand.Reader, data)
	
	headers := cose.Headers{
		Protected: cose.ProtectedHeader{
			cose.HeaderLabelAlgorithm: cose.AlgorithmRS256,
		},
	}
	sig, _ = cose.Sign1(rand.Reader, cs, headers, data, nil)
	
}	
```

### CODE REVIEW

File: signer.go [hunk 1]
```diff
@@ -88,6 +88,8 @@ func NewSigner(alg Algorithm, key crypto.Signer) (Signer, error) {
 		return &ed25519Signer{
 			key: key,
 		}, nil
+	case AlgorithmRS256, AlgorithmRS384, AlgorithmRS512:
```
[the remark this answers is not in the document]
Comment (hslatman) [reply signer.go]: @qmuntal changed that in 0eac3fc75f513da8a08e9b8094b3837f74fac73c. Now all algorithms will return either the textual name or the numeric identifier, similar to "unknown algorithm value -1", suffixed with the `ErrAlgorithmNotSupported`.

**Review by qmuntal — APPROVED** [review]
Thanks! LGTM

**Comment by SteveLasker** [comment]
@or13, PTAL

**Review by OR13 — APPROVED** [review]
The error message could be better.

Algorithm not supported,  external signer required (link to documentation).

As is, the user is likely to assume it's not possible to use the library with the algorithm... Unless they read the comments in the code.

**Comment by hslatman** [comment]
> The error message could be better.
> 
> Algorithm not supported, external signer required (link to documentation).
> 
> As is, the user is likely to assume it's not possible to use the library with the algorithm... Unless they read the comments in the code.

I think you mean the general error message; not just for the `RSx` constants? I agree that the error could be improved. That might need more valid COSE algorithm constants to be added, so that the error message can distinguish between valid algorithm identifiers (that require an external signer) vs. invalid algorithm identifiers (that won't/shouldn't work with external signers)

**Review by roywill — APPROVED** [review]

**Pull request merged**